include/onnxruntime/core/common/const_pointer_container.h

```diff
@@ -20,17 +20,41 @@ class ConstPointerContainer {
   class ConstIterator {
    public:
     using const_iterator = typename Container::const_iterator;
+    using iterator_category = std::input_iterator_tag;
+    using value_type = T*;
```

Comment:
not sure myself here - should value_type = const T*? and pointer = const T**, reference = const T*&?

Reply:
Matches onnx/graph_node_list.h (and we are enforcing the addition of the 'const'), so I believe it's correct. .

```
struct iterator_traits<ONNX_NAMESPACE::generic_graph_node_list_iterator<T>> {
  using difference_type = int64_t;
  using value_type = T*;
  using pointer = T**;
  using reference = T*&;
  using iterator_category = bidirectional_iterator_tag;
};
```